packages/commonwealth/client/scripts/views/pages/manage_community/ManageCommunityPage.tsx

```diff
@@ -57,48 +70,51 @@ const ManageCommunityPage = () => {
     setMods(memberMods);
   };
 
-  const searchMembers = async (searchQuery?: string) => {
-    try {
-      let profiles = [];
-
-      const cachedResult = membersCache.get(searchQuery);
-      if (cachedResult) {
-        profiles = cachedResult.profiles;
-      } else {
-        const res = await axios.get(`${app.serverUrl()}/searchProfiles`, {
-          params: {
-            chain: app.activeChainId(),
-            search: searchQuery || '',
-            page_size: 100,
-            page: 1,
-            include_roles: true,
-          },
-        });
-        if (res.data.status !== 'Success') {
-          throw new Error('Could not fetch members');
-        }
-        membersCache.set(searchQuery, res.data.result);
-        profiles = res.data.result.profiles;
-      }
-
-      let roles = [];
+  const fetchSearchResults = async ({ pageParam = 0 }) => {
+    const {
+      data: { result },
+    } = await axios.get<{ result: ProfilesSearchResponse }>(`/api/profiles`, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      params: {
+        chain: app.activeChainId(),
+        search: searchTerm,
+        limit: (50).toString(),
+        page: pageParam.toString(),
+        order_by: orderBy,
+        order_direction: orderDirection,
+        include_roles: 'true',
+      },
+    });
+    return result.results;
+  };
 
-      if (profiles.length > 0) {
-        roles = profiles.map((profile) => {
-          return {
-            ...(profile.roles[0] || {}),
-            Address: profile.addresses[0],
-            id: profile.addresses[0].id,
-          };
-        });
-      }
-      setRoleData(roles);
-      setInitialized(true);
-    } catch (err) {
-      setRoleData([]);
-      setInitialized(true);
+  const { data: searchResults, refetch } = useQuery({
+    queryKey: [
+      'search-members-manage',
+      {
+        debouncedSearchTerm,
+        chain: app.activeChainId(),
+        orderBy,
+        orderDirection,
+      },
+    ],
+    queryFn: fetchSearchResults,
+  });
+
+  const roleData = useMemo(() => {
+    if (!searchResults) {
+      return [];
     }
-  };
+    return searchResults.map((profile) => {
+      return {
+        ...(profile.roles[0] || {}),
+        Address: profile.addresses[0],
+        id: profile.addresses[0].id,
+      };
+    });
+  }, [searchResults]);
```

Comment:
this is really great that we try to introduce react query in more places. But we should try to follow the approach that already has been established for the react-query. Take a look on `state/api/comments` or `state/api/topics`. We should not have all this logic located in the component because it is for displaying data, not for business logic. 

You guys have done great job tiding up the server and move routes into the `resources` so let's now try to clear out the front end in similar way.

So what I would recommend is create new directory `state/api/profiles` and inside each file should be separate endpoint. Then in the component you import just a query and trigger it whenever necessary.

Reply:
Thanks for the examples. Will follow.